File: aldebaran-common/aldebaran-redis/src/main/java/com/aldebaran/redis/RedisConfiguration.java
package com.aldebaran.redis;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Conditional;
import org.springframework.context.annotation.Configuration;
import redis.clients.jedis.JedisPool;
import redis.clients.jedis.JedisPoolConfig;


@Configuration
@Conditional(RedisEnabledCondition.class)
public class RedisConfiguration {

    @Bean
    public JedisPoolConfig jedisPoolConfig(RedisProperties redisProperties) {
        JedisPoolConfig jedisPoolConfig = new JedisPoolConfig();
        jedisPoolConfig.setMaxTotal(redisProperties.getPoolSize());
        return jedisPoolConfig;
    }

    @Bean
    public JedisPool jedisPool(RedisProperties redisProperties,
                               JedisPoolConfig jedisPoolConfig) {
        return  new JedisPool(jedisPoolConfig,
                              redisProperties.getRedisHost(),
                              redisProperties.getRedisPort());
    }
}

File: aldebaran-auth-micro/aldebaran-auth-server/src/main/java/com/aldebaran/auth/server/config/RedisConfiguration.kt
package com.aldebaran.auth.server.config

import org.springframework.context.annotation.Condition
import org.springframework.context.annotation.ConditionContext
import org.springframework.context.annotation.Conditional
import org.springframework.context.annotation.Configuration
import org.springframework.core.type.AnnotatedTypeMetadata


@Configuration
@Conditional(RedisConfiguration.RedisCondition::class)
open class RedisConfiguration {

    class RedisCondition : Condition {
        override fun matches(context: ConditionContext?,
                             metadata: AnnotatedTypeMetadata?): Boolean {
            return "true" == context?.environment?.getProperty("redis.enabled")
        }
    }
}